src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs

```diff
@@ -323,43 +304,33 @@ public void SetCapability(CompletionCapability capability, ClientCapabilities cl
             };
         }
 
+        // TODO: Unwrap this, it's confusing as it doesn't cover all cases and we conditionally
+        // override it when it's inaccurate.
         private static CompletionItemKind MapCompletionKind(CompletionType completionType)
         {
-            switch (completionType)
+            return completionType switch
             {
-                case CompletionType.Command:
-                    return CompletionItemKind.Function;
-
-                case CompletionType.Property:
-                    return CompletionItemKind.Property;
-
-                case CompletionType.Method:
-                    return CompletionItemKind.Method;
-
-                case CompletionType.Variable:
-                case CompletionType.ParameterName:
-                    return CompletionItemKind.Variable;
-
-                case CompletionType.File:
-                    return CompletionItemKind.File;
-
-                case CompletionType.Folder:
-                    return CompletionItemKind.Folder;
-
-                default:
-                    return CompletionItemKind.Text;
-            }
+                CompletionType.Unknown => CompletionItemKind.Text,
+                CompletionType.Command => CompletionItemKind.Function,
+                CompletionType.Method => CompletionItemKind.Method,
+                CompletionType.ParameterName => CompletionItemKind.Variable,
+                CompletionType.ParameterValue => CompletionItemKind.Value,
+                CompletionType.Property => CompletionItemKind.Property,
+                CompletionType.Variable => CompletionItemKind.Variable,
+                CompletionType.Namespace => CompletionItemKind.Module,
+                CompletionType.Type => CompletionItemKind.TypeParameter,
+                CompletionType.Keyword => CompletionItemKind.Keyword,
+                CompletionType.File => CompletionItemKind.File,
+                CompletionType.Folder => CompletionItemKind.Folder,
+                CompletionType.History => CompletionItemKind.Reference,
+                CompletionType.Text => CompletionItemKind.Text,
+                _ => throw new NotImplementedException(),
```

Comment:
```suggestion
                _ => throw new ArgumentOutOfRangeException(nameof(completionType)),
```

`NotImplementedException` is usually for interface implementations that are explicitly not implemented.